### ISSUE TITLE
$Shape<T> allows undefined and null values

### ISSUE THREAD

**Issue by mhagmajer** [facebook/flow#5884]
```js
/* @flow */

(undefined: $Shape<{}>); // no error
(null: $Shape<{ a: 1 }>); // no error either
```

I know that [`$Shape<T>`](https://github.com/facebook/flow/blob/v0.66.0/src/typing/type_annotation.ml#L289) is sort of an undocumented feature, yet I doubt this is expected behaviour.

[Checked against 0.66](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAKArgOwCYCmUAljgXgFxgAkAygBYCGADgQDwDeAvgHwCUAbkw4sMGFTpNWnMIyoBGML0FA)

**Comment by AlexanderShushunov** [comment]
v0.85 - still works incorrect.

**Comment by AlexanderShushunov** [comment]
Workaround https://flow.org/try/#0C4TwDgpgBAJAygQwG4TgCwZAPAFQHxQC8s6mEWA3lAgFxQB2ArgLYBGEATlAL4EBkUCtwDcAKADGAe3oBnYFABmkyUSgAKCgm514yVBmxVaDFuy68AlEQJDRopZI3GATNwv3lGtx8cVWdV3cHNUZ6ABMIBQBLeggwoM8mABsk9yA

**Comment by Bannerets** [comment]
fixed in v0.90.0
